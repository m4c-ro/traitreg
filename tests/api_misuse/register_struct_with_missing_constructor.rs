fn main() {}



trait MyTrait {}

struct MyStruct;

#[traitreg::register(new)]
impl MyTrait for MyStruct {}
