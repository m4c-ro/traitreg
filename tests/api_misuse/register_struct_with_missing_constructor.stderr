error[E0599]: no function or associated item named `new` found for struct `MyStruct` in the current scope
 --> tests/api_misuse/register_struct_with_missing_constructor.rs:9:22
  |
7 | struct MyStruct;
  | --------------- function or associated item `new` not found for this struct
8 |
9 | #[traitreg::register(new)]
  |                      ^^^ function or associated item not found in `MyStruct`
